### PR TITLE
Prioritize the use of dependModules when both dependModules and main are satisfied

### PR DESCRIPTION
**Pull request by weixsun**
Closes gh-24

### CODE REVIEW

File: initializer-generator/src/main/java/com/alibaba/initializer/generation/extension/build/maven/MulitModuleMavenBuildProjectContributor.java [hunk 1]
```diff
@@ -152,22 +153,19 @@ public void contribute(Path projectRoot) throws IOException {
...
-                List<Module> subModules = arch.getSubModules();
-                for (Module subModule : subModules) {
-                    if (subModule == module) {
-                        continue;
-                    }
-                    addModuleDependency(subModule.getName());
+                if (dependModules.isEmpty()) {
+                    // main module depend all other submodules
+                    List<String> subModules = arch.getSubModules().stream().distinct()
+                            .dropWhile(subModule -> subModule == module)
```
Comment (theonefx) [inline initializer-generator/src/main/java/com/alibaba/initializer/generation/extension/build/maven/MulitModuleMavenBuildProjectContributor.java:161]: filter(subModule -> subModule != module)

Comment (weixsun) [reply initializer-generator/src/main/java/com/alibaba/initializer/generation/extension/build/maven/MulitModuleMavenBuildProjectContributor.java]: Done

**Review by theonefx — COMMENTED** [review]
LGTM

**Review by theonefx — APPROVED** [review]
LGTM

**Pull request merged**